### PR TITLE
[VL] Support regr_slope aggregate function

### PR DESCRIPTION
**Pull request by liujiayi771**
## What changes were proposed in this pull request?

Support regr_slope aggregate function.

## How was this patch tested?

Add new test case in `VeloxAggregateFunctionsSuite`.

### CODE REVIEW

**Comment by liujiayi771** [comment]
cc @rui-mo @PHILO-HE, thanks.

File: backends-velox/src/main/scala/org/apache/gluten/execution/HashAggregateExecTransformer.scala [hunk 1]
```diff
@@ -146,7 +146,7 @@ abstract class HashAggregateExecTransformer(
           val (sparkOrders, sparkTypes) =
             aggFunc.aggBufferAttributes.map(attr => (attr.name, attr.dataType)).unzip
           val veloxOrders = VeloxIntermediateData.veloxIntermediateDataOrder(aggFunc)
-          val adjustedOrders = sparkOrders.map(veloxOrders.indexOf(_))
+          val adjustedOrders = sparkOrders.map(VeloxIntermediateData.getAttrIndex(veloxOrders, _))
```
Comment (rui-mo) [inline backends-velox/src/main/scala/org/apache/gluten/execution/HashAggregateExecTransformer.scala:149]: Is this change a bug fix?

**Review by rui-mo — COMMENTED** [review]
Would you resolve the conflicts? Thanks.

File: backends-velox/src/main/scala/org/apache/gluten/execution/HashAggregateExecTransformer.scala (hunk 1 above)
Comment (liujiayi771) [reply backends-velox/src/main/scala/org/apache/gluten/execution/HashAggregateExecTransformer.scala:149]: This change is to support another situation.
> Agg functions with inconsistent ordering of intermediate data between Velox and Spark. The
   strings in the Seq comes from the aggBufferAttributes of Spark's aggregate function, and they
   are arranged in the order of fields in Velox's Accumulator. The reason for using a
   two-dimensional Seq is that in some cases, a field in Velox will be mapped to multiple
   Attributes in Spark's aggBufferAttributes. For example, the fourth field of Velox's RegrSlope
   Accumulator is mapped to both xAvg and avg in Spark's RegrSlope aggBufferAttributes. In this
   scenario, when passing the output of Spark's partial aggregation to Velox, we only need to
   take one of them.

`VeloxIntermediateData.getAttrIndex` is used to get the index from a two-dimensional Seq.

**Review by rui-mo — APPROVED** [review]
Thanks!

**Comment by PHILO-HE** [comment]
@liujiayi771, please note `checkOperatorMatch` has been changed to `checkGlutenOperatorMatch`.

**Review by PHILO-HE — APPROVED** [review]

**Comment by liujiayi771** [comment]
@PHILO-HE CH backend has not fixed this yet.
```
ERROR] /home/jenkins/agent/workspace/gluten/gluten-ci/gluten/backends-clickhouse/src/test/scala/org/apache/gluten/execution/GlutenClickHouseTPCHSaltNullParquetSuite.scala:1159: error: not found: value checkOperatorMatch
[2024-04-02T13:11:00.599Z] [ERROR]     runQueryAndCompare(sql)(checkOperatorMatch[ProjectExecTransformer])
[2024-04-02T13:11:00.599Z] [ERROR]                             ^
[2024-04-02T13:11:01.551Z] [ERROR] /home/jenkins/agent/workspace/gluten/gluten-ci/gluten2/backends-clickhouse/src/test/scala/org/apache/gluten/execution/GlutenClickHouseTPCHSaltNullParquetSuite.scala:1159: error: not found: value checkOperatorMatch
[2024-04-02T13:11:01.564Z] [ERROR]     runQueryAndCompare(sql)(checkOperatorMatch[ProjectExecTransformer])
[2024-04-02T13:11:01.564Z] [ERROR]                             ^
[2024-04-02T13:11:02.125Z] [ERROR] one error found
[2024-04-02T13:11:02.380Z] [ERROR] exception compilation error occurred!!!
```

**Pull request merged**